### PR TITLE
ST-136: API Endpoints for tax code history and tax code change in year

### PR DESCRIPTION
**Pull request by amaalali**

### CODE REVIEW

File: test/uk/gov/hmrc/tai/service/TaxCodeChangeServiceImplSpec.scala [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+class TaxCodeChangeServiceImplSpec extends PlaySpec with MockitoSugar {
+
+  "hasTaxCodeChanged" should {
+    "return true" when {
+      "there has been a tax code change the year" in {
+        val thisYear = LocalDate.now()
+        val lastYear = LocalDate.now().minusYears(1)
+
+        val taxCodeHistory = TaxCodeHistory(
+          nino = "AA000000",
```
Comment (lewey) [inline test/uk/gov/hmrc/tai/service/TaxCodeChangeServiceImplSpec.scala]: Nino

File: test/uk/gov/hmrc/tai/service/TaxCodeChangeServiceImplSpec.scala [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        val service = new TaxCodeChangeServiceImpl(mockConnector)
+        Await.result(service.hasTaxCodeChanged(testNino), 5.seconds) mustEqual true
+      }
+    }
+
+    "return false" when {
+      "there are no tax code changes" in {
+
+        val taxCodeHistory = TaxCodeHistory(
+          "AA000000",
```
Comment (lewey) [inline test/uk/gov/hmrc/tai/service/TaxCodeChangeServiceImplSpec.scala]: Nino

File: test/uk/gov/hmrc/tai/service/TaxCodeChangeServiceImplSpec.scala [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+        val service = new TaxCodeChangeServiceImpl(mockConnector)
+        Await.result(service.hasTaxCodeChanged(testNino), 5.seconds) mustEqual false
+      }
+
+      "there has not been a tax code change in the year" in {
+
+        val lastYear = LocalDate.now().minusYears(1)
+        val taxCodeHistory = TaxCodeHistory(
+          "AA000000",
```
Comment (lewey) [inline test/uk/gov/hmrc/tai/service/TaxCodeChangeServiceImplSpec.scala]: Nino

File: app/uk/gov/hmrc/tai/service/TaxCodeChangeService.scala [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import uk.gov.hmrc.tai.connectors.TaxCodeChangeConnector
+import uk.gov.hmrc.tai.model.TaxCodeHistory
+import uk.gov.hmrc.tai.model.tai.TaxYear
+import uk.gov.hmrc.tai.util.TaiConstants
+
+import scala.concurrent.Future
+
+class TaxCodeChangeServiceImpl @Inject()(taxCodeChangeConnector: TaxCodeChangeConnector) extends TaxCodeChangeService {
+
+  def hasTaxCodeChanged(nino: Nino): Future[Boolean] = {
```
[the remark this answers is not in the document]
Comment (amaalali) [reply app/uk/gov/hmrc/tai/service/TaxCodeChangeService.scala]: Addressed lower down where the URL is formed

**Pull request merged**